### PR TITLE
libtree: init at 2.0.0

### PR DESCRIPTION
**Pull request by prusnak**
###### Motivation for this change

- add new useful tool called libtree (and its dependencies) - https://github.com/haampie/libtree

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rardiol** [comment]
All 3 packages have tests that can be enabled. These should enable them:

For elfio
```
+  buildInputs = [ boost ];
+
+  cmakeFlags = [ "-DELFIO_BUILD_TESTS=ON" ];
+
+  doCheck = true;
+
```

For termcolor
```
  nativeBuildInputs = [ cmake ];

  cmakeFlags = [ "-DTERMCOLOR_TESTS=ON" ];

  doCheck = true;

  checkPhase = ''
    ./test_termcolor
  '';
```

For libtree
```
  cmakeFlags = [ "-DLIBTREE_BUILD_TESTS=ON" ];

  doCheck = true;

  nativeBuildInputs = [ cmake gtest ];
```

Termcolor license is inaccurate, it should be bsd3: https://github.com/ikalnytskyi/termcolor/blob/master/LICENSE 

> https://github.com/ikalnytskyi/termcolor

> It's licensed under the BSD (3-clause) License. That basically means: do whatever you want as long as copyright sticks around. 

And this makes the bundling feature use nix's chrpath and strip instead of $PATH. DIdn't seem to deal well with nixos though.
```
  prePatch = ''
    substituteInPlace src/main.cpp \
      --replace "std::string strip = \"strip\";" "std::string strip = \"${binutils.out}/bin/strip\";" \
      --replace "std::string chrpath = \"chrpath\";" "std::string chrpath = \"${chrpath.out}/bin/chrpath\";"
  '';
```

**Comment by prusnak** [comment]
@rardiol Thank you for the stellar review. I addressed all your points and force pushed.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
met.maintainer is missing everywhere.

File: pkgs/development/tools/misc/libtree/default.nix [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    (fetchpatch {
+      url = "https://github.com/haampie/libtree/commit/219643ff6edcae42c9546b8ba38cfec9d19b034e.patch";
+      sha256 = "sha256-vdFmmBdBiOT3QBcwd3SuiolcaFTFAb88kU1KN8229K0=";
+    })
+  ];
+
+  postPatch = ''
+    substituteInPlace src/main.cpp \
+      --replace "std::string strip = \"strip\";" "std::string strip = \"${binutils.out}/bin/strip\";" \
+      --replace "std::string chrpath = \"chrpath\";" "std::string chrpath = \"${chrpath.out}/bin/chrpath\";"
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/libtree/default.nix:37]:
```suggestion
      --replace "std::string strip = \"strip\";" "std::string strip = \"${binutils}/bin/strip\";" \
      --replace "std::string chrpath = \"chrpath\";" "std::string chrpath = \"${chrpath}/bin/chrpath\";"
```
Isn't the right path chosen?

File: pkgs/development/libraries/termcolor/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  nativeBuildInputs = [ cmake ];
+
+  cmakeFlags = [ "-DTERMCOLOR_TESTS=ON" ];
+
+  doCheck = true;
+
+  checkPhase = ''
+    ./test_termcolor
+  '';
```
Comment (r-burns) [inline pkgs/development/libraries/termcolor/default.nix:26]:
```suggestion
  checkPhase = ''
    runHook preCheck
    ./test_termcolor
    runHook postCheck
  '';
```

Comment (prusnak) [reply pkgs/development/libraries/termcolor/default.nix]: Fixed in 86d00441b0e59e9d915b4ffd675e2a68642aa3da

File: pkgs/development/tools/misc/libtree/default.nix (hunk 1 above)
Comment (prusnak) [reply pkgs/development/tools/misc/libtree/default.nix]: Fixed in 840624332a3ffa8b909a061c03ac58855034f48f

**Comment by rardiol** [comment]
Feel free to add myself as maintainer

**Comment by prusnak** [comment]
> Feel free to add myself as maintainer

Added in c549f68b7dcfb44bd386954df779c3ffb4397408

**Pull request merged**